### DISCUSSION TITLE
Migration matrices

### DISCUSSION THREAD

**Discussion by alanrogers** [tskit-dev/msprime#2178] [Q&A] [answered]
I'm setting up a migration matrix for 2-D stepping stone migration on a torus. It seems to work right, but the demography debugger says that the diagonal entries of the matrix are 0. I was expecting a positive fraction, so that the entries of each row would sum to 1. Is this as expected?

**Comment by jeromekelleher** [comment, marked as answer]
This is expected, as the rate of deme-to-itself migration is required to be zero (possibly not for any particularly good reasons). 

Does this help?

**Reply by alanrogers** [reply]
Yes. Thanks.
